code/datums/controllers/process/world.dm

```diff
@@ -29,6 +29,13 @@
 	doWork()
 		last_object = "sun.calc_position"
 		sun.calc_position()
+		z2sun.calc_position()
+		z3sun.calc_position()
+		z5sun.calc_position()
+		earth_sun.calc_position()
+		io_sun.calc_position()
+		senex_sun.calc_position()
+		solarium_sun.calc_position()
```

Comment:
What's the performance impact here?

Reply:
I imagine it's quite high, sadly. Haven't tested it yet.
I'm going to throw in some extra vars and things so that large chunks of the calc_position code gets skipped under certain circumstances, because I don't think all of the azone suns need to actually calculate rotation, they can probably just update their solars/trackers once and then be ignored. The many `for` loops in the calc_position() code worry me a lot. I've been also considering staggering the `counter`s on the `calc_position` calculation so that they're not all done at the same time and possibly stop a lag spike from occurring every two seconds exactly.

Reply:
for now i've made all the azone suns not update their angular positions ever. Should help. Will be testing it with and without those optimisations later.

Reply:
okay so this is it running with 8 suns
![image](https://user-images.githubusercontent.com/51260013/229471012-4a5a547c-3aad-402a-9ef4-43d46c96b6e2.png)
I don't know if this is terrible but it doesn't look too bad to me
![image](https://user-images.githubusercontent.com/51260013/229471352-99f919f3-7d5f-40b2-b08e-915c478b6244.png)

Reply:
That's extremely laggy.

Reply:
shit

Reply:
![image](https://user-images.githubusercontent.com/51260013/230865826-b54d11a5-f30b-4f04-b409-e22176cae11f.png)
Okay so this is it under standard conditions, only the z1 sun turning, everything else just being skipped.